lib/model/model.go

```diff
@@ -180,11 +182,41 @@ func (m *Model) StartFolderRW(folder string) {
 		p.versioner = versioner
 	}
 
+	m.warnAboutOverwritingConfigs(folder)
+
 	m.Add(p)
 
 	l.Okln("Ready to synchronize", folder, "(read-write)")
 }
 
+func (m *Model) warnAboutOverwritingConfigs(folder string) {
+	if m.folderCfgs[folder].ReadOnly {
+		return
+	}
+
+	folderLocation := m.folderCfgs[folder].Path()
+	ignores := m.folderIgnores[folder]
+
+	var filesAtRisk []string
+	for _, protectedFilePath := range m.protectedConfigFiles {
+		// check if file is synced in this folder
+		if !strings.HasPrefix(protectedFilePath, folderLocation) {
+			continue
+		}
+
+		// check if file is ignored
+		if ignores.Match(protectedFilePath) {
+			continue
+		}
+
+		filesAtRisk = append(filesAtRisk, protectedFilePath)
+	}
+
+	if len(filesAtRisk) > 0 {
+		l.Warnln("The following configuration files may be overwritten from other nodes. See http://docs.syncthing.net/users/config.html#syncing-configuration-files for more information.", strings.Join(filesAtRisk, ", "))
```

Comment:
Needs a colon at the end of the sentence, and again I think we should talk about "protected files" or just "files" and not config files here. The link will annoyingly not be clickable when this is shown in the GUI. Perhaps the error showing thing there should be improved to auto detect links...